src/fermat.jl

```diff
@@ -0,0 +1,39 @@
+
+using NumberTheoreticTransforms
+
+export fermat
+
+"""
+    fermat(convolved, kernel)
+
+Calculates deconvolution with Number Theoretic Transform.
+"""
+function fermat(convolved::AbstractArray{T, 1}, h::AbstractArray{T, 1}, g::T, q::T) where {T<:Integer}
+    N = length(convolved)
+    bias = div(q-1, 2) + 1 #negative numbers are represeted by the upper half of [0, q) range
+    H = fnt(h, g, q)
+    H_inv = invmod.(H, q)
+    D = fft(h) |> prod |> real |> T
+    Dm = mod(D, q)
+    xm = ifnt(H_inv .* fnt(convolved, g, q), g, q)
+    x0 = mod.(Dm * xm, q)
+    x0[x0 .>= bias] .-= q
+    x = x0
+    y_prim = fft(h) .* fft(x0) |> ifft .|> real .|> round.|> T
+    @assert rem.(convolved * D - y_prim, q) == zeros(T, N)
+    y = div.(convolved * D - y_prim, q)
+    ym = mod.(y, q)
+    j = 1
+    while y != zeros(T, length(convolved))
+        xj = ifnt(H_inv .* fnt(ym, g, q), g, q)
+        xj[xj .>= bias] .-= q
+        x = x .+ (xj * q^j)
+        y_prim = fft(h) .* fft(xj) |> ifft .|> real .|> round .|> T
+        @assert rem.(y - y_prim, q) == zeros(T, N)
```

Comment:
Also here, I think this is equivalent to
```julia
all(iszero, rem.(y - y_prim, q))
```
without allocating a new array.

Reply:
> > The biggest limitation of this method is that each element of H matrix needs to have inverse, what is hard to achieve for real data, so I'm not sure if Deconvolution.jl is good place for algorithms like this.
> 
> While currently implemented methods are probably easier to use in practice, I don't see why this would be a problem for including this method here 😉

I was just concerned about introducing dependencies to LinearAlgebra.jl, and SpecialMatrices.jl but I managed to remove them finally. Also I am not able to perform 512x512 2D deconvolution due to zeros in H matrix, so I thought it won't be useful. Maybe I will find sensible 1D usage example related to audio data or quantitative finance.